File: Easy/007_lowercase/C#/LowerCase.cs
using System;
using System.IO;
using System.Linq;

class LowerCase
{
    class Lower
    {
        ///<summary>
        /// Returns the string in lower case
        ///</summary>    
        public string ToLower(string word)
        {
            string loweredWord = "";
            foreach (char ch in word)
                loweredWord += GetLower(ch);
            return loweredWord;
        }

        /// <summar>
        /// Returns the char in lowercase
        /// </summary>
        private char GetLower(char c)
        {
            // if the char is in between A-Z, return lowered version
            if (c >= 'A' && c <= 'Z')
                return (char)(c+32);
            else
                return c;
        }
    }

    public static void Main (string[] args)
    {
        // for each line in the file
        foreach (string line in File.ReadLines(args[0]))
        {
            // init a new lower object
            Lower lower = new Lower();

            // write to stdout the lin in lower case
            Console.WriteLine(lower.ToLower(line));
        }
    }
}

File: Easy/007_lowercase/Java/LowerCase.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;

public class LowerCase
{

    public static void main(String[] args) throws IOException
    {
        ArrayList<String> list = new ArrayList<>();
        if (args.length > 0)
        {
            // create a reader to read the file
            BufferedReader reader = new BufferedReader(new FileReader(args[0]));
    
            // init a string to hold each line
            String line = "";
    
            // while able to get a line from the file, add to the list
            while ((line = reader.readLine()) != null)
                list.add(line);
    
            // close out the reader
            reader.close();
    
            // for each item in the list
            for (String item : list)
            {
                // print to stdout, the lower case version of the string
                System.out.println(item.toLowerCase());
            }
        }

    }

}